deepinv/models/bilateral.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+import torch
+from torch import Tensor
+from .base import Denoiser
+
+
+class BilateralFilter(Denoiser):
+    r"""
+    Bilateral filter.
+
+    The bilateral filter as it was introduced in :footcite:t:`tomasi1998bilateral`
+    where each output pixel is a normalized weighted average of neighboring pixels in a spatial window.
+    The weights factor into a spatial kernel and a range kernel:
+
+        \hat{x}_i = (1 / W_i) * \sum_{j \in \omega(i)} k_d(i - j) k_r(x_i - x_j) x_j
```

Comment:
```suggestion
    .. math::

        \hat{x}_i = (1 / W_i) * \sum_{j \in \omega(i)} k_d(i - j) k_r(x_i - x_j) x_j
```

This is needed for the equation to be displayed properly